scripts/deconvolution_funs.R

```diff
@@ -0,0 +1,179 @@
+  # parse together from vep output "Protein_position" and "Amino_acid"
+  #' input: []_sarscov2_parsed.txt, parsed vcf from VEP CLI, comma separated
+  #' this function is for parsing the information about mutation position in the
+  #' amino acid sequence, the reference aa and the alternative mutation into the
+  #' aa-mutation-notation which is later on comparable to the lists of signature
+  #' mutations
+
+  # reading in whole vep txt output
+  # you should include in the vep script to parse out the #
+  # in the beginning of the line or include that step here.
+  # TODO: include check about correct VEP file input format
+  vepfile_df <- fread(vepfile, sep = ",", header = TRUE)
+  # parsing snv and protmut location
+
+
+  locations <- vepfile_df %>%
+
+    rename(
+      prot_pos  = Protein_position,
+      conseq    = Consequence,
+      gene_name = SYMBOL) %>%
+
+    # get general info on mutation and its position
+    mutate(
+      mut_nucs   = str_extract(Uploaded_variation, "[A-Z*]/[A-Z*]$"),
+      mut_chrom  = str_extract(Location, "^[^:]+"),
+      mut_start  = str_extract(Location, "(?<=:)[0-9]+[^-]"),
+      mut_end    = str_extract(Location, "(?<=-)[0-9]+")) %>%
+
+    # generate unique string describing mutation
+    mutate(
+      mut_ref = str_extract(mut_nucs, "^[A-Z*]"),
+      mut_var = str_extract(mut_nucs,  "[A-Z*]$"),
+
+      mut_str = paste0(mut_ref, mut_start, mut_var)
+    ) %>%
+
+    # get infos on mutation consequences for protein
+    mutate(
+      # TODO This may behave unexpectedly; in the case of no change aa_ref
+      # and aa_var are the same, aa_var will not be empty / NA
+      aa_ref = str_extract(Amino_acids, "^[A-Z*]"),
+      aa_var = str_extract(Amino_acids,  "[A-Z*]$"),
+
+      aa_str = paste0(aa_ref, prot_pos, aa_var)
+    ) %>%
+
+    # delete all rows with no protein position value
+    # TODO Why do we delete these mutations? This just means they do not
+    # affect the proteins, the mutations may still be informative...
+    filter(! str_detect(prot_pos, "^-")) %>%
+    # TODO What is the relevance of this information?
+    # specific B117 mutations: 21990-21993, 21764-21770, maybe also 3675-3677,
+    # 69-70 - all there
+
+    # remove unneeded cols
+    dplyr::select(-matches(names(vepfile_df), ignore.case = FALSE)) %>%
+
+    # This also removes variants that essentially look the same, but differ in
+    # their VEP determined Gene
+    # e.g. Gene:ENSSASG00005000002 <> Gene:ENSSASG00005000003
+    distinct()
+
+    # TODO Previously deletions were handled by generating pseudo rows for each
+    # one. I currently do not have access to a file with deletions, and so
+    # cannot understand how to handle this properly. However this solution here
+    # is still better, because due to a bug the previous solution could almost
+    # never run. Now there is at least a warning...
+    if (any(!is.na(locations$mut_end))) {
+      warning("Found a deletion. Help.")
```

Comment:
The dangling issue of how to deal with deletions in `deconvolution.R` is fixed on my local branch btw. I will push once the review is through, or I am prompted to do so.

Reply:
Oops, during some reshuffling of my local repo I didn't realize this PR would be affected and pushed onto the branch. The result should be that commit "simplify deconvolution" now has additional subcommits relating to this point, see below.
```
  remove unused detectable_deletions function

    I looked through the entire results section of the project dir
    and could find no files with "deletions" in them...

  remove calculation of lowecase others col

    This was supposed to already be gone before I worked on this.
    It was commented out, but when I copied stuff around I didn't
    realize it and activated it again.

  format deconvolution_funs.R

  add proper handling of indels to get_protein_mut

    This replaces the detectable_deletions function and works both for
    insertions and deletions. Plus it actually runs.

  fix variable in deconvolution_funs.R

    Missed it during renaming.

  update regex for nuc muts to include indels

    It is conjecture whether this works, but if amino acids have - for
    deletions, nucleotides probably do too.

  remove browser() debugging call i forgot

  fix concat of replacement rows in get_protein_mut

    Missed this earlier, didnt even test it then.

  fix deletion of indel rows in get_protein_mut
```